src/modeler/AutoRest.Swagger/Validation/SkuModelValidation.cs

```diff
@@ -62,5 +72,34 @@ public override bool IsValid(Dictionary<string, Schema> definitions)
 
             return true;
         }
+
+        /// <summary>
+        /// Evaluates if the reference is of the provided data type.
+        /// </summary>
+        /// <param name="reference">reference to evaluate</param>
+        /// <param name="definitions">definition list</param>
+        /// <param name="dataType">Datatype value to evaluate</param>
+        /// <returns>true if the reference is of the provided data type. False otherwise.</returns>
+        private bool evaluateIf(string reference, Dictionary<string, Schema> definitions, Model.DataType dataType)
+        {
+            if (reference == null)
+            {
+                return false;
+            }
+
+            string definitionName = reference.Substring(reference.LastIndexOf('/') + 1).Trim();
+            Schema schema = definitions.GetValueOrNull(definitionName);
+            if (schema == null)
+            {
+                return false;
+            }
+
+            if (schema.Type == dataType || (schema.Type == null && schema.Reference != null && evaluateIf(schema.Reference, definitions, dataType)))
```

Comment:
should we allow another reference in this case? or should it the type be defined at this level? Since this is specific to sku model, I'm wondering if it'd make sense to have another reference.